--- transmission-1.72/configure.ac.orig	2009-06-17 02:01:56.000000000 +0200
+++ transmission-1.72/configure.ac	2009-06-18 18:25:44.000000000 +0200
@@ -81,6 +81,14 @@
 AC_HEADER_STDC
 AC_HEADER_TIME
 
+AC_MSG_CHECKING([for SYS_fallocate])
+AC_TRY_LINK([#include <sys/syscall.h>
+#include <sys/types.h>],
+            [return syscall(SYS_fallocate, 0, 0, (loff_t) 0, (loff_t) 0);],
+            [AC_DEFINE([HAVE_SYS_FALLOCATE],[1],[Defined if you have the SYS_fallocate syscall number])
+             AC_MSG_RESULT([yes])],
+            [AC_MSG_RESULT([no])])
+
 AC_CHECK_FUNCS([lrintf strlcpy daemon dirname basename daemon strcasecmp localtime_r posix_fallocate memmem])
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
--- transmission-1.72/libtransmission/fdlimit.c.orig	2009-06-17 02:01:44.000000000 +0200
+++ transmission-1.72/libtransmission/fdlimit.c	2009-06-18 18:24:45.000000000 +0200
@@ -43,6 +43,11 @@
  #include <fcntl.h>
 #endif
 
+#ifdef HAVE_SYS_FALLOCATE
+ #include <linux/falloc.h>
+ #include <sys/syscall.h>
+#endif
+
 #ifdef HAVE_XFS_XFS_H
  #include <xfs/xfs.h>
 #endif
@@ -176,6 +181,16 @@
             success = !fcntl( fd, F_PREALLOCATE, &fst );
         }
 # endif
+# ifdef HAVE_SYS_FALLOCATE
+        if( !success )
+        {
+            success = !syscall( SYS_fallocate, fd, FALLOC_FL_KEEP_SIZE, 0, (loff_t)length );
+            if( !success )
+            {
+                success = !syscall( SYS_ftruncate, fd, (loff_t)length );
+            }
+        }
+# endif
 # ifdef HAVE_POSIX_FALLOCATE
         if( !success )
         {
