--- transmission-2.61/configure.ac.orig
+++ transmission-2.61/configure.ac
@@ -3,8 +3,8 @@
 dnl         "Z" for unsupported trunk builds,
 dnl         "0" for stable, supported releases
 dnl these should be the only two lines you need to change
-m4_define([user_agent_prefix],[2.80])
-m4_define([peer_id_prefix],[-TR2800-])
+m4_define([user_agent_prefix],[2.42])
+m4_define([peer_id_prefix],[-TR2420-])
 
 AC_INIT([transmission],
         [user_agent_prefix],
