diff -ur transmission-2.00-orig//qt/details.cc transmission-2.00/qt/details.cc
--- transmission-2.00-orig//qt/details.cc	2010-06-17 09:10:47.000000000 -0600
+++ transmission-2.00/qt/details.cc	2010-06-18 17:02:37.000000000 -0600
@@ -143,6 +143,7 @@
     t->addTab( w = createOptionsTab( ),   tr( "Options" ) );
     myWidgets << w;
     layout->addWidget( t );
+    resize(600, height());
 
     QDialogButtonBox * buttons = new QDialogButtonBox( QDialogButtonBox::Close, Qt::Horizontal, this );
     connect( buttons, SIGNAL(rejected()), this, SLOT(close()));
@@ -1158,7 +1159,7 @@
     v->addWidget( myPeerTree, 1 );
 
     const QFontMetrics m( font( ) );
-    QSize size = m.size( 0, "1024 MiB/s" );
+    QSize size = m.size( 0, "1024,0 MiB/s" );
     myPeerTree->setColumnWidth( COL_UP, size.width( ) );
     myPeerTree->setColumnWidth( COL_DOWN, size.width( ) );
     size = m.size( 0, " 100% " );
Only in transmission-2.00/qt: details.cc~
Only in transmission-2.00/qt: details.cc.orig
