diff -ur transmission-2.00-orig//configure.ac transmission-2.00/configure.ac
--- transmission-2.00-orig//configure.ac	2010-06-17 10:45:12.000000000 -0600
+++ transmission-2.00/configure.ac	2010-06-18 17:05:50.000000000 -0600
@@ -21,16 +21,7 @@
 AM_INIT_AUTOMAKE([1.9 tar-ustar])
 AC_PROG_LIBTOOL
 
-if test m4_substr(peer_id_prefix,6,1) = "0"; then
-  supported_build=yes
-  CPPFLAGS="$CPPFLAGS -DNDEBUG"
-else
-  supported_build=no
-  if test "x$GCC" = "xyes" ; then
-    CFLAGS="$CFLAGS -g -O0"
-    CXXFLAGS="$CXXFLAGS -g -O0"
-  fi
-fi
+CPPFLAGS="$CPPFLAGS -DNDEBUG"
 AM_CONDITIONAL(TR_UNSTABLE, test "x$supported_build" = "xno")
 
 ##
