diff -burNp transmission-1.91/configure.ac transmission-1.91-dud/configure.ac
--- transmission-1.91/configure.ac	2010-02-22 03:42:55.000000000 +0100
+++ transmission-1.91-dud/configure.ac	2010-02-23 10:37:08.076245513 +0100
@@ -108,6 +108,14 @@ fi
 AC_HEADER_STDC
 AC_HEADER_TIME
 
+AC_MSG_CHECKING([for SYS_fallocate])
+AC_TRY_LINK([#include <sys/syscall.h>
+#include <sys/types.h>],
+		[return syscall(SYS_fallocate, 0, 0, (loff_t) 0, (loff_t) 0);],
+		[AC_DEFINE([HAVE_SYS_FALLOCATE],[1],[Defined if you have the SYS_fallocate syscall number])
+		AC_MSG_RESULT([yes])],
+		[AC_MSG_RESULT([no])])
+
 AC_CHECK_FUNCS([pread pwrite lrintf strlcpy daemon dirname basename strcasecmp localtime_r fallocate64 posix_fallocate memmem strtold syslog valloc getpagesize posix_memalign])
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
diff -burNp transmission-1.91/libtransmission/fdlimit.c transmission-1.91-dud/libtransmission/fdlimit.c
--- transmission-1.91/libtransmission/fdlimit.c	2010-02-22 03:42:48.000000000 +0100
+++ transmission-1.91-dud/libtransmission/fdlimit.c	2010-02-23 10:40:51.375812134 +0100
@@ -43,6 +43,10 @@
  #include <fcntl.h>
 #endif
 
+#ifdef HAVE_SYS_FALLOCATE
+ #include <sys/syscall.h>
+#endif
+
 #ifdef HAVE_FALLOCATE64
   /* FIXME can't find the right #include voodoo to pick up the declaration.. */
   extern int fallocate64( int fd, int mode, uint64_t offset, uint64_t len );
@@ -179,6 +183,12 @@ preallocateFileFull( const char * filena
             success = !fcntl( fd, F_PREALLOCATE, &fst );
         }
 # endif
+# ifdef HAVE_SYS_FALLOCATE
+	if( !success )
+	{
+	    success = !syscall( SYS_fallocate, fd, 0, (loff_t) 0, (loff_t) length );
+	}
+# endif
 # ifdef HAVE_POSIX_FALLOCATE
         if( !success )
         {
