--- a/qt/details.cc
+++ b/qt/details.cc
@@ -165,6 +165,7 @@
     t->addTab( w = createOptionsTab( ),   tr( "Options" ) );
     myWidgets << w;
     layout->addWidget( t );
+    resize(640, height());
 
     QDialogButtonBox * buttons = new QDialogButtonBox( QDialogButtonBox::Close, Qt::Horizontal, this );
     connect( buttons, SIGNAL(rejected()), this, SLOT(close()));
@@ -1277,7 +1278,7 @@
     v->addWidget( myPeerTree, 1 );
 
     const QFontMetrics m( font( ) );
-    QSize size = m.size( 0, "1024 MiB/s" );
+    QSize size = m.size( 0, " 1024,0 MiB/s" );
     myPeerTree->setColumnWidth( COL_UP, size.width( ) );
     myPeerTree->setColumnWidth( COL_DOWN, size.width( ) );
     size = m.size( 0, " 100% " );
