--- transmission-1.83/configure.ac.orig	2010-02-07 14:31:56.000000000 +0100
+++ transmission-1.83/configure.ac	2010-02-07 14:31:56.000000000 +0100
@@ -81,6 +81,14 @@
 AC_HEADER_STDC
 AC_HEADER_TIME
 
+AC_MSG_CHECKING([for SYS_fallocate])
+AC_TRY_LINK([#include <sys/syscall.h>
+#include <sys/types.h>],
+            [return syscall(SYS_fallocate, 0, 0, (loff_t) 0, (loff_t) 0);],
+            [AC_DEFINE([HAVE_SYS_FALLOCATE],[1],[Defined if you have the SYS_fallocate syscall number])
+             AC_MSG_RESULT([yes])],
+            [AC_MSG_RESULT([no])])
+
 AC_CHECK_FUNCS([pread pwrite lrintf strlcpy daemon dirname basename strcasecmp localtime_r posix_fallocate memmem strtold syslog])
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
--- transmission-1.83/libtransmission/fdlimit.c.orig	2010-01-29 02:01:32.000000000 +0100
+++ transmission-1.83/libtransmission/fdlimit.c	2010-02-07 05:53:09.000000000 +0100
@@ -43,6 +43,10 @@
  #include <fcntl.h>
 #endif
 
+#ifdef HAVE_SYS_FALLOCATE
+ #include <sys/syscall.h>
+#endif
+
 #ifdef HAVE_XFS_XFS_H
  #include <xfs/xfs.h>
 #endif
@@ -168,6 +172,12 @@
             success = !fcntl( fd, F_PREALLOCATE, &fst );
         }
 # endif
+# ifdef HAVE_SYS_FALLOCATE
+        if( !success )
+        {
+            success = !syscall( SYS_fallocate, fd, 0, (loff_t) 0, (loff_t) length );
+        }
+# endif
 # ifdef HAVE_POSIX_FALLOCATE
         if( !success )
         {
